packages/tables/src/styled/StyledHeaderRow.ts

```diff
@@ -5,37 +5,61 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import styled, { ThemeProps, DefaultTheme } from 'styled-components';
+import styled, { ThemeProps, DefaultTheme, css } from 'styled-components';
 import { math } from 'polished';
-import { retrieveComponentStyles, getColorV8, DEFAULT_THEME } from '@zendeskgarden/react-theming';
-import { StyledBaseRow, IStyledRowProps } from './StyledRow';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import { StyledBaseRow } from './StyledBaseRow';
 import { StyledOverflowButton } from './StyledOverflowButton';
+import { ITableProps } from '../types';
 
 const COMPONENT_ID = 'tables.header_row';
 
-const getHeaderRowHeight = (props: IStyledRowProps & ThemeProps<DefaultTheme>) => {
-  if (props.size === 'large') {
+interface IStyledHeaderRowProps {
+  $size?: ITableProps['size'];
+}
+
+const getHeaderRowHeight = (props: IStyledHeaderRowProps & ThemeProps<DefaultTheme>) => {
+  if (props.$size === 'large') {
     return `${props.theme.space.base * 18}px`;
-  } else if (props.size === 'small') {
+  } else if (props.$size === 'small') {
     return `${props.theme.space.base * 10}px`;
   }
 
   return `${props.theme.space.base * 12}px`;
 };
 
+const colorStyles = ({ theme }: ThemeProps<DefaultTheme>) => {
+  return css`
+    border-bottom-color: ${getColor({ variable: 'border.default', theme })};
+  `;
+};
+
+const sizeStyles = (props: ThemeProps<DefaultTheme>) => {
+  const rowHeight = getHeaderRowHeight(props);
+
+  return css`
+    height: ${rowHeight};
+    vertical-align: bottom;
+
+    ${StyledOverflowButton} {
+      margin-top: 0;
+      margin-bottom: calc(${math(`${rowHeight} / 2`)} - 1em);
+    }
+  `;
+};
+
 export const StyledHeaderRow = styled(StyledBaseRow).attrs({
   'data-garden-id': COMPONENT_ID,
   'data-garden-version': PACKAGE_VERSION
-})`
-  border-bottom-color: ${props => getColorV8('neutralHue', 300, props.theme)};
-  height: ${getHeaderRowHeight};
-  vertical-align: bottom;
+})<IStyledHeaderRowProps>`
   font-weight: ${props => props.theme.fontWeights.semibold};
 
+  ${colorStyles}
+
+  ${sizeStyles}
```

Comment:
```suggestion
  ${sizeStyles}

  ${colorStyles}
```

[nit] per API docs, these should be switched .. although it shouldn't happen, color should hold the more significant position if there happens to be a cascade.